src/graph_notebook/magics/streams.py

```diff
@@ -0,0 +1,225 @@
+import re
+import json
+import urllib.request
+import urllib.error
+import ipywidgets as widgets
+import queue
+from IPython.display import display, HTML, clear_output
+from graph_notebook.neptune.client import STREAM_AT, STREAM_AFTER, STREAM_TRIM, STREAM_EXCEPTION_NOT_FOUND
+class EventId:
+    def __init__(self, commit_num=1, op_num=1):
+        self.commit_num = int(commit_num)
+        self.op_num = int(op_num)
+        
+    
+    def update(self, event_id):
+        if event_id is not None:
+            self.commit_num = event_id.commit_num
+            self.op_num = event_id.op_num
+            
+    def duplicate(self):
+        return EventId(self.commit_num, self.op_num)
+    
+    def value(self):
+        return '{}/{}'.format(self.commit_num, self.op_num)
+
+class StreamClient:
+    
+    def __init__(self, wb_client, uri_with_port, limit=10):
+        self.wb_client = wb_client
+        self.uri_with_port = uri_with_port
+        self.limit=limit
+    
+    
+    def get_events(self, language, event_id, iterator):
+        try:
+            jsonresponse = self.wb_client.stream(self.__stream_uri(language),
+                                                 iteratorType = iterator,
+                                                 commitNum = event_id.commit_num,
+                                                 opNum = event_id.op_num,
+                                                 limit=self.limit)
+                                                 
+            if 'records' in jsonresponse:
+                records = jsonresponse['records']
+                first_event = EventId(records[0]['eventId']['commitNum'], records[0]['eventId']['opNum'])
+                last_event = EventId(jsonresponse['lastEventId']['commitNum'], jsonresponse['lastEventId']['opNum'])
+
+                return (records, first_event, last_event)
```

Comment:
Nit: Redundant parentheses, remove; same on lines 49 and 52

Reply:
Fixed